module/util/util.go

```diff
@@ -6,6 +6,33 @@ import (
 	"github.com/onflow/flow-go/module"
 )
 
+// DefaultFallbackStrategy simple fallback strategy assuming clients are stored in a list
+// this strategy will manage the active client index
+type DefaultFallbackStrategy struct {
+	numOfClients int
+	clientIndex  int
```

Comment:
I suggest we protect this with a mutex. The two use-cases we have now are single-threaded, but it's easy to imagine uses of this component which would concurrently access a client index and invoke `Failure` from two separate threads.